packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -248,3 +221,103 @@ def wait_for_container_response():
         endpoints = dcos_api_session.marathon.get_app_service_endpoints(marathon_config['id'])
         assert len(endpoints) == 1, 'The marathon app should have been deployed exactly once.'
         test_containers(endpoints)
+
+
+def test_metrics_containers_nan(dcos_api_session):
+    """Assert that the metrics API can handle app metric gauges with NaN values."""
+    task_name = 'test-metrics-containers-nan'
+    metric_name = 'test_metrics_containers_nan'
+    marathon_app = {
+        'id': '/' + task_name,
+        'instances': 1,
+        'cpus': 0.1,
+        'mem': 128,
+        'cmd': '\n'.join([
+            'echo "Sending gauge with NaN value to $STATSD_UDP_HOST:$STATSD_UDP_PORT"',
+            'echo "{}:NaN|g" | nc -w 1 -u $STATSD_UDP_HOST $STATSD_UDP_PORT'.format(metric_name),
+            'echo "Done. Sleeping forever."',
+            'while true; do',
+            '  sleep 1000',
+            'done',
+        ]),
+        'container': {
+            'type': 'MESOS',
+            'docker': {'image': 'library/alpine'}
+        },
+        'networks': [{'mode': 'host'}],
+    }
+
+    @retrying.retry(wait_fixed=(2 * 1000), stop_max_delay=(150 * 1000))
+    def _get_app_metric_datapoint_value_for_task(dcos_api_session, node: str, task_name: str, dp_name: str):
+        app_metrics = get_app_metrics_for_task(dcos_api_session, node, task_name)
+        assert app_metrics is not None, "missing metrics for task {}".format(task_name)
+        dps = [dp for dp in app_metrics['datapoints'] if dp['name'] == dp_name]
+        assert len(dps) == 1, 'expected 1 datapoint for metric {}, got {}'.format(dp_name, len(dps))
+        return dps[0]['value']
+
+    with dcos_api_session.marathon.deploy_and_cleanup(marathon_app, check_health=False):
+        endpoints = dcos_api_session.marathon.get_app_service_endpoints(marathon_app['id'])
+        assert len(endpoints) == 1, 'The marathon app should have been deployed exactly once.'
+        node = endpoints[0].host
+
+        # NaN should be converted to empty string.
+        metric_value = _get_app_metric_datapoint_value_for_task(dcos_api_session, node, task_name, metric_name)
```

Comment:
Did we finally decide that empty string was the right value for NaN?

Reply:
Yeah